src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/ECDiffieHellman.cs

```diff
@@ -433,5 +434,173 @@ public virtual unsafe bool TryExportECPrivateKey(Span<byte> destination, out int
                 }
             }
         }
+
+        /// <summary>
+        /// Imports an RFC 7468 PEM-encoded key, replacing the keys for this object.
+        /// </summary>
+        /// <param name="input">The PEM text of the key to import.</param>
+        /// <exception cref="ArgumentException">
+        /// <paramref name="input"/> does not contain a PEM-encoded key with a recognized label.
+        ///   <para>
+        ///       -or-
+        ///   </para>
+        /// <paramref name="input"/> contains multiple PEM-encoded keys with a recognized label.
+        ///   <para>
+        ///       -or-
+        ///   </para>
+        /// <paramref name="input"/> contains an encrypted PEM-encoded key.
+        /// </exception>
+        /// <remarks>
+        ///   <para>
+        ///   Unsupported or malformed PEM-encoded objects will be ignored. If multiple supported PEM labels
+        ///   are found, an exception is raised to prevent importing a key when
+        ///   the key is ambiguous.
+        ///   </para>
+        ///   <para>
+        ///   This method supports the following PEM labels:
+        ///   <list type="bullet">
+        ///     <item><description>PUBLIC KEY</description></item>
+        ///     <item><description>PRIVATE KEY</description></item>
+        ///     <item><description>EC PRIVATE KEY</description></item>
+        ///   </list>
+        ///   </para>
+        /// </remarks>
+        public override void ImportFromPem(ReadOnlySpan<char> input)
+        {
+            PemKeyImportHelpers.ImportPem(input, label => {
+                if (label.SequenceEqual("PRIVATE KEY"))
+                {
+                    return ImportPkcs8PrivateKey;
+                }
+                else if (label.SequenceEqual("PUBLIC KEY"))
```

Comment:
Since these literals are spread out across multiple files, I think we should go ahead and create a PemConstants/PemHeaders/PemLabels constants file.